crates/typst/src/geom/color.rs

```diff
@@ -334,183 +954,696 @@ cast! {
 }
 
 /// A weight for color mixing.
-struct Weight(f32);
+struct Weight(f64);
 
 cast! {
     Weight,
-    v: f64 => Self(v as f32),
-    v: Ratio => Self(v.get() as f32),
+    v: f64 => Self(v),
+    v: Ratio => Self(v.get()),
 }
 
 /// Convert an RGBA color to four components in the given color space.
-fn rgba_to_vec4(color: RgbaColor, space: ColorSpace) -> [f32; 4] {
+pub fn color_to_vec4(color: Color, space: ColorSpace) -> [f64; 4] {
```

Comment:
The comment seems outdated. It's also a bit confusing because a `to_vec4` function exist in `ColorExt`. What's the difference? This converts to a space first? Shouldn't that be two different steps?